apps/notifications-service/src/notifications/notifications.service.ts

```diff
@@ -0,0 +1,252 @@
+      throw error;
+    }
+  }
+
+  async sendPushNotification(notification: Notification) {
+    try {
+      // Get user devices from user service
+      const userServiceUrl = getRequiredEnvVar('USER_SERVICE_URL');
+      const response = await fetch(`${userServiceUrl}/users/${notification.userId}/devices`);
+      
+      if (!response.ok) {
+        this.logger.error(`Failed to fetch user devices: ${response.statusText}`);
+        return;
+      }
+
+      const devices = await response.json() as any[];
+      const activeDevices = devices.filter((device: any) => device.isActive);
+      
+      if (activeDevices.length === 0) {
+        this.logger.log(`No active devices found for user ${notification.userId}`);
+        return;
+      }
+
+      const messages: ExpoPushMessage[] = activeDevices
+        .filter((device: any) => device.deviceType === 'MOBILE')
+        .map((device: any) => ({
+          to: device.deviceToken,
+          sound: 'default',
+          title: notification.title,
+          body: notification.body,
+          data: {
+            notificationId: notification.id,
+            type: notification.type,
+            swapId: notification.swapId,
+          },
+          priority: 'high',
+          channelId: 'swap-notifications',
+        }));
+
+      await this.sendExpoPushNotifications(messages, notification.id);
+    } catch (error) {
+      this.logger.error('Failed to send push notification', error);
+    }
+  }
+
+  async sendPushNotificationToDevice(
+    deviceToken: string, 
+    title: string, 
+    body: string, 
+    data?: PushNotificationData
+  ): Promise<boolean> {
+    try {
+      if (!Expo.isExpoPushToken(deviceToken)) {
+        this.logger.warn(`Invalid Expo push token: ${deviceToken}`);
+        return false;
+      }
+
+      const message: ExpoPushMessage = {
+        to: deviceToken,
+        sound: 'default',
+        title,
+        body,
+        data: data || {},
+        priority: 'high',
+        channelId: 'swap-notifications',
+      };
+
+      const tickets = await this.sendExpoPushNotifications([message]);
+      return tickets.length > 0;
+    } catch (error) {
+      this.logger.error('Failed to send push notification to device', error);
+      return false;
+    }
+  }
+
+  async sendPushNotificationToUser(
+    userId: string, 
+    title: string, 
+    body: string, 
+    data?: PushNotificationData
+  ): Promise<boolean> {
+    try {
+      // Get user devices from user service
+      const userServiceUrl = getRequiredEnvVar('USER_SERVICE_URL');
+      const response = await fetch(`${userServiceUrl}/users/${userId}/devices`);
+      
+      if (!response.ok) {
+        this.logger.error(`Failed to fetch user devices: ${response.statusText}`);
+        return false;
+      }
+
+      const devices = await response.json() as any[];
+      const activeDevices = devices.filter((device: any) => device.isActive);
```

Comment:
and then remove any in lambdas across the board

Reply:
Done